nix/modules/lanzaboote.nix

```diff
@@ -7,6 +7,12 @@ let
   };
 
   configurationLimit = if cfg.configurationLimit == null then 0 else cfg.configurationLimit;
+  timeout = if config.boot.loader.timeout == null then 0 else config.boot.loader.timeout;
```

Comment:
Does this mean no menu at all when I don't set this option?

The systemd-boot behavior now is that if I don't set the timeout option explicitly, it will also not generate a timeout line in the config, which sounds more user-friendly.

Reply:
From reading the docs, 0 seems to be the default. So if there is no timeout line the value should still be 0: https://www.freedesktop.org/software/systemd/man/loader.conf.html#timeout

Also, the Nix default is 5: https://search.nixos.org/options?channel=22.11&show=boot.loader.timeout&from=0&size=50&sort=relevance&type=packages&query=boot.loader.timeout
A user would need to explicitly configure the timeout to be 0.

So no, if you don't set this option it will be 5. If you explicitly set it to `null` it will be 0.

Reply:
I think this is not valid, with respect to the documentation of the NixOS option. The `then` branch should be `menu-force` (like what nixpkgs does for systemd-boot).

(I had to recently discover this was not properly honored across all bootloaders in nixpkgs unfortunately, but someone made PRs to uniformize stuff.)

Reply:
In general, I'd argue there should be some utility to map timeouts to proper values according to the bootloader…

Reply:
That `null` would lead to an indefinite menu is not intuitive at all. `null != 0` would be very confusing for users IMO.

And indeed systemd-boot in Nixpkgs is NOT implemented that way. `null` leads to no `timeout` line at all in the config. See https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix#L23 and https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py#L65

So all in all the behaviour implemented in this PR is the same as the behaviour of systemd-boot in Nixpkgs.

I agree with the utility, but we should land this PR before we think about long-term design as it solves a very sharp edge for users (and is consitent with behaviour users expect from systemd-boot).